### PR TITLE
pkg/config: Mark Gateway.Name & Gateway.Namespace as deprecated

### PR DESCRIPTION
**Pull request by stevesloka**
The fields in the Gateway configuration, Name & Namespace, are deprecated and will be removed in Contour v1.18.

Updates #3617

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: site/content/docs/main/configuration.md
Comment (skriss) [inline site/content/docs/main/configuration.md:183]: For consistency:
```suggestion
| name (Deprecated) | string | contour | DEPRECATED: This field specifies the name of a Gateway.  |
```

**Review by skriss — COMMENTED** [review]
One nit, otherwise looks reasonable and inline with our [config file deprecation policy](https://projectcontour.io/resources/deprecation-policy/)

**Review by skriss — APPROVED** [review]

**Review by youngnick — APPROVED** [review]
Should we hold this until we merge #3797 though? Since we haven't actually technically agreed until that happens?

**Comment by stevesloka** [comment]
>  Since we haven't actually technically agreed until that happens?

Yeah maybe? It's already there: https://github.com/projectcontour/contour/blob/main/pkg/config/parameters.go#L293-L296

**Comment by youngnick** [comment]
I think we're agreed that we will do this, we're just cleaning up the doc now, so it's probably fine.

**Comment by danehans** [comment]
cc: @nak3

**Pull request merged**
